client/nginx.go

```diff
@@ -43,25 +45,31 @@ func NewNginxClient(httpClient *http.Client, apiEndpoint string) (*NginxClient,
 
 // GetStubStats fetches the stub_status metrics.
 func (client *NginxClient) GetStubStats() (*StubStats, error) {
-	resp, err := client.httpClient.Get(client.apiEndpoint)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, client.apiEndpoint, nil)
```

Comment:
what is the rationale for using `NewRequestWithContext`?

Reply:
Coming from the new linter

> You should use noctx if sending http request in your library. Passing context.Context enables library user to cancel http request, getting trace information and so on.

Reply:
I don't think we use `cancel http request, getting trace information` in our Prometheus exporter, so not fully sure if we need it.  perhaps it makes sense to be a bit pragmatic here and introduce the change after we find a use case for it?

Reply:
Removed the timeout for now.